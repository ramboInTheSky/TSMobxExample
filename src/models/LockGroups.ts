export class LockGroup{
    constructor(id?: string, name?: string, type?: string){
        this.id = id || ''
        this.name = name || ''
        this.type = type || 'Room'
    }
    id: string = ''
    name: string = ''
    type: string = ''
    notes: string = ''
    selected: boolean = false
}

export class CreateLockGroup{
    name: string = ''
    type: string = ''
    notes: string = ''
}

export interface LockGroupResponse {
	count: number
	filters: Array<string>
	items: Array<LockGroup>
}
