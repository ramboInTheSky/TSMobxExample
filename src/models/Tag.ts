export class Tag{
    tagNumber: string | null = null

    constructor(tag?: string){
        this.tagNumber = tag || null
    }
}