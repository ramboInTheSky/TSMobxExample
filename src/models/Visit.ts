import { LockGroup } from './LockGroups'
export class Visit {
    id: string = ''
    lockGroups?: Array<LockGroup | string>  = []
    fromDate: string = ''
    toDate: string = ''
    tags: Array<string> = []
    archetypeIds: Array<string> = []
}

export class CreateVisit {
    lockGroups?: Array<LockGroup> = []
    fromDate: string = ''
    toDate: string = ''
    tags: Array<string> = []
    archetypeIds: Array<string> = []
}

export interface Archetype {
    id: string
    name: string
    primary: boolean
    lockGroupNames: Array<string>
}

export class Archetypes {
    primary: Array<Archetype> = []
    secondary: Array<Archetype> = []
}