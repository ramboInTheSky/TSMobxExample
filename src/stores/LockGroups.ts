import { observable, action } from 'mobx'
import { Store } from './APIStore'
import { LockGroup } from '../models/LockGroups'
import urls from '../constants/urls'

const sortingFn = (a: LockGroup, b: LockGroup) => {
	const first = a.name
	const last = b.name
	if (first > last) return -1
	if (last > first) return 1
	return 0
}

export class LockGroupStore extends Store<LockGroup> {
	@observable
	public selectedLockGroups: Array<LockGroup> = []

	constructor() {
		super(urls.lockGroups)
	}

	async getList(filter?: string) {
		const data = await super.getList(undefined, filter)
		this.data =
			data && data.items ? data.items : []
	}

	@action 
	setList(newList: Array<LockGroup>){
		this.data = newList
	}

}
