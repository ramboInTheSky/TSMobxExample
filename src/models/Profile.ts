import { Visit } from './Visit'

export type ProfileType = 'visitor' | 'member'

export class Profile {
	id?: string = undefined
	fullName?: string = undefined
	firstName: string = ''
	lastName: string = ''
	company?: string = undefined
	mobileNumber?: string = ''
	emailAddress?: string = ''
	notes?: string = undefined
	created?: string = undefined
	type?: ProfileType = undefined
	visits?: Array<Visit> = undefined
}

export class CreateProfile {
	firstName: string = ''
	lastName: string = ''
	company?: string = undefined
	mobileNumber?: string = ''
	emailAddress?: string = ''
	notes?: string = undefined
	type?: ProfileType = undefined
}

export class EditProfile {
	id?: string = undefined
	firstName: string = ''
	lastName: string = ''
	company?: string = undefined
	mobileNumber?: string = ''
	emailAddress?: string = ''
	notes?: string = undefined
	created?: string = undefined
	type?: ProfileType = undefined
}

export interface ProfileResponse {
	count: number
	filters: Array<string>
	items: Array<Profile>
}
