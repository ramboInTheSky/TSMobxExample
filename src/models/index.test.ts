import { Profile, CreateProfile, EditProfile } from './Profile'
import { Visit, CreateVisit } from './Visit'
import { Tag } from './Tag';
import { LockGroup } from './LockGroups';

describe('Profile', () => {
	it('renders correctly', () => {
		expect(new Profile()).toMatchSnapshot()
	})
})
describe('CreateProfile', () => {
	it('renders correctly', () => {
		expect(new CreateProfile()).toMatchSnapshot()
	})
})
describe('EditProfile', () => {
	it('renders correctly', () => {
		expect(new EditProfile()).toMatchSnapshot()
	})
})
describe('Visit', () => {
	it('renders correctly', () => {
		expect(new Visit()).toMatchSnapshot()
	})
})
describe('CreateVisit', () => {
	it('renders correctly', () => {
		expect(new CreateVisit()).toMatchSnapshot()
	})
})
describe('Tag', () => {
	it('renders correctly', () => {
		expect(new Tag()).toMatchSnapshot()
	})
})
describe('LockGroup', () => {
	it('renders correctly', () => {
		expect(new LockGroup()).toMatchSnapshot()
	})
})