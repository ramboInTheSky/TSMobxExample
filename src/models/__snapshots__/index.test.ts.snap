// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfile renders correctly 1`] = `
CreateProfile {
  "company": undefined,
  "emailAddress": "",
  "firstName": "",
  "lastName": "",
  "mobileNumber": "",
  "notes": undefined,
  "type": undefined,
}
`;

exports[`CreateVisit renders correctly 1`] = `
CreateVisit {
  "archetypeIds": Array [],
  "fromDate": "",
  "lockGroups": Array [],
  "tags": Array [],
  "toDate": "",
}
`;

exports[`EditProfile renders correctly 1`] = `
EditProfile {
  "company": undefined,
  "created": undefined,
  "emailAddress": "",
  "firstName": "",
  "id": undefined,
  "lastName": "",
  "mobileNumber": "",
  "notes": undefined,
  "type": undefined,
}
`;

exports[`LockGroup renders correctly 1`] = `
LockGroup {
  "id": "",
  "name": "",
  "notes": "",
  "selected": false,
  "type": "Room",
}
`;

exports[`Profile renders correctly 1`] = `
Profile {
  "company": undefined,
  "created": undefined,
  "emailAddress": "",
  "firstName": "",
  "fullName": undefined,
  "id": undefined,
  "lastName": "",
  "mobileNumber": "",
  "notes": undefined,
  "type": undefined,
  "visits": undefined,
}
`;

exports[`Tag renders correctly 1`] = `
Tag {
  "tagNumber": null,
}
`;

exports[`Visit renders correctly 1`] = `
Visit {
  "archetypeIds": Array [],
  "fromDate": "",
  "id": "",
  "lockGroups": Array [],
  "tags": Array [],
  "toDate": "",
}
`;
